root/edit/details/AddRelationship.js

```diff
@@ -36,7 +38,13 @@ const AddRelationship = ({edit}: Props): React.MixedElement => {
     <table className="details add-relationship">
       <tr>
         <th>{l('Relationship:')}</th>
-        <td><Relationship allowNew relationship={relationship} /></td>
+        <td>
+          <Relationship
+            allowNewEntity0={edit.display_data.entity0_is_new}
+            allowNewEntity1={edit.display_data.entity1_is_new}
```

Comment:
You should be able to to check `edit.data.entity0.id` etc. directly here, I think, which would make adding `entityX_is_new` to the display data unnecessary.

Reply:
... That would be sensible, yes. Done!